application-task-test/application-task-test-docker/src/test/it/com/xwiki/task/test/ui/NotificationIT.java

```diff
@@ -0,0 +1,379 @@
+        // by GreenMail running on the host.
+        3025
+    }, extensionOverrides = {
+        @ExtensionOverride(
+            extensionId = "com.google.code.findbugs:jsr305",
+            overrides = {
+                "features=com.google.code.findbugs:annotations"
+            }
+        )
+    }, properties = {
+        // The scheduler UI needs programming rights
+        "xwikiPropertiesAdditionalProperties=test.prchecker.excludePattern=.xwiki:Scheduler\\.WebHome",
+        // The Mail module contributes a Hibernate mapping that needs to be added to hibernate.cfg.xml
+        "xwikiDbHbmCommonExtraMappings=mailsender.hbm.xml,notification-filter-preferences.hbm.xml",
+        // Add the Scheduler plugin used by Mail Resender Scheduler Job
+        "xwikiCfgPlugins=com.xpn.xwiki.plugin.scheduler.SchedulerPlugin"
+    }, extraJARs = {
+        // It's currently not possible to install a JAR contributing a Hibernate mapping file as an Extension. Thus,
+        // we need to provide the JAR inside WEB-INF/lib. See https://jira.xwiki.org/browse/XWIKI-19932
+        "org.xwiki.platform:xwiki-platform-notifications-filters-default:14.10",
+        // The Solr store is not ready yet to be installed as an extension, so we need to add it to WEB-INF/lib
+        // manually. See https://jira.xwiki.org/browse/XWIKI-21594
+        "org.xwiki.platform:xwiki-platform-eventstream-store-solr:14.10",
+        // It's currently not possible to install a JAR contributing a Hibernate mapping file as an Extension. Thus
+        // we need to provide the JAR inside WEB-INF/lib. See https://jira.xwiki.org/browse/XWIKI-19932
+        "org.xwiki.platform:xwiki-platform-mail-send-storage:14.10",
+        // The Scheduler plugin needs to be in WEB-INF/lib since it's defined in xwiki.properties and plugins are loaded
+        // by XWiki at startup, i.e. before extensions are provisioned for the tests
+        "org.xwiki.platform:xwiki-platform-scheduler-api:14.10"
+    }, resolveExtraJARs = true
+)
+public class NotificationIT
+{
+    @RegisterExtension
+    static GreenMailExtension mail = new GreenMailExtension(ServerSetupTest.SMTP);
+
+    private static String TEST_USERNAME = "NotificationTestUser";
+
+    private static String TEST_EDITOR_USERNAME = "NotificationTestEditor";
+
+    private static String PASSWORD = "password";
+
+    private final DocumentReference TASK_MACRO_PAGE =
+        new DocumentReference("xwiki", "Main", "TestTaskMacroNotifications");
+
+    private final String TASK_MACRO_CONTENT =
+        "{{task reference=\"Task_0\" createDate=\"2024/12/20 15:13\" status=\"Done\" reporter=\"\"}}My task{{mention reference=\""
+            + TEST_USERNAME + "\" anchor=\"sss-QXfgr\" style=\"FULL_NAME\"/}}{{/task}}";
+
+    private String TEST_TASK_NAME = "Test Task";
+
+    private String TEST_PROJECT_NAME = "Test Project";
+
+    @AfterAll
```

Comment:
Semantically, I feel it's more correct to have `AfterAll` after the `Before*` methods, to have them in order